File: api/src/main/java/com/github/av2/api/model/Product.java
package com.github.av2.api.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Product information")
public class Product {
    @Schema(description = "The unique identifier for the product", required = true)
    private Integer productId;
    
    @Schema(description = "The ID of the supplier for this product", required = true)
    private Integer supplierId;
    
    @Schema(description = "The name of the product", required = true)
    private String name;
    
    @Schema(description = "Description of the product")
    private String description;
    
    @Schema(description = "The price of the product", required = true)
    private Float price;
    
    @Schema(description = "The SKU (Stock Keeping Unit) of the product", required = true)
    private String sku;
    
    @Schema(description = "The unit type (e.g., piece, kg, etc.)", required = true)
    private String unit;
    
    @Schema(description = "The image filename for the product")
    private String imgName;
    
    @Schema(description = "The discount percentage for the product (0.0 to 1.0)")
    private Float discount;
}

File: frontend/src/types/product.ts
export interface IProduct {
  productId: number;
  supplierId: number;
  name: string;
  description: string;
  price: number;
  sku: string;
  unit: string;
  imgName: string;
  discount: number | null;
}

export const getDiscountedPrice = (product: Pick<IProduct, 'price' | 'discount'>): number => {
  const discount = product.discount ?? 0;
  return product.price * (1 - discount);
};
